src/sortStudents.ts

```diff
@@ -1,16 +1,62 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverageValue(arr: number[]): number {
+  return arr.reduce((calc: number, num: number): number => calc + num, 0)
+    / arr.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyStudents: Student[] = [...students];
+
+  copyStudents.sort((first: Student, second: Student): number => {
+    switch (sortBy) {
+      case SortType.Name:
+        return (order === 'asc')
+          ? first[sortBy].localeCompare(second[sortBy])
+          : second[sortBy].localeCompare(first[sortBy]);
+
+      case SortType.Surname:
+        return (order === 'asc')
+          ? first[sortBy].localeCompare(second[sortBy])
+          : second[sortBy].localeCompare(first[sortBy]);
+
+      case SortType.Age:
+        return (order === 'asc')
+          ? +first[sortBy] - +second[sortBy]
+          : +second[sortBy] - +first[sortBy];
+
+      case SortType.Married:
+        return (order === 'asc')
+          ? +first[sortBy] - +second[sortBy]
+          : +second[sortBy] - +first[sortBy];
```

Comment:
as it's the same you can use multiple cases
```suggestion
      case SortType.Married:
      case SortType.Age:
        return (order === 'asc')
          ? +first[sortBy] - +second[sortBy]
          : +second[sortBy] - +first[sortBy];
```